### DISCUSSION TITLE
Get PnP search results to return translated variants of SharePoint site pages

### DISCUSSION THREAD

**Discussion by HenningStrand** [microsoft-search/pnp-modern-search#1885] [Q&A] [answered]
We have site languages activated on some SharePoint publishing sites where news is translated into a set number of languages. By default it looks like the out-of-box News rollup web part and Highlighted contents web parts filter these site pages showing the variant that corresponds to the users language preference/browser language. 

How would we do similar in the PnP search results web part? With our current query all site pages that match our managed properties based search query are returned. For translated site pages that means seeing the same one in multiple translations. What is needed to be added to get only the user's language variants displayed, or default to English if no other variant is published? Thanks for tips!

**Comment by wobba** [comment, marked as answer]
Take a look at https://www.techmikael.com/2020/09/search-and-modern-multilingual-pages-in.html and the solution is at the end of that blog post :)

**Reply by HenningStrand** [reply]
Thanks a lot, Mikael! Quick and to the point. :)
